File: Formula1/src/Formula1/JAVA/F2.java
package Formula1.JAVA;

public class F2 extends Monoplaza{
    private boolean superLicencia;

        ////////////////CONSTRUCTOR/////////////
    public F2(String nombrePiloto, String nacionalidad, int numeroPlaza, String escuderia, int cantidadPuntos, boolean pSuperLicencia) {
        super(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos);
        this.superLicencia = pSuperLicencia;
    }

    ///////////////////GETTERS Y SETTERS/////////////////
    public void setSuperLicencia(boolean superLicencia) {
        this.superLicencia = superLicencia;
    }
    public boolean isSuperLicencia() {
        return superLicencia;
    }
    ////////////////METODOS//////////////////
    @Override
    public boolean posicionValida(int posicion) {
        if(posicion >=1 && posicion<=24){
        return true;
    }else{
        return false;
    } 
    }

    public void otorgarPuntos(int posicion, boolean vueltaRapida) {
    if (!posicionValida(posicion)) return;

    switch (posicion) {
        case 1 -> cantidadPuntos += 10;
        case 2 -> cantidadPuntos += 8;
        case 3 -> cantidadPuntos += 7;
        case 4 -> cantidadPuntos += 6;
        case 5 -> cantidadPuntos += 5;
        case 6 -> cantidadPuntos += 4;
        case 7 -> cantidadPuntos += 3;
        case 8 -> cantidadPuntos += 2;
        case 9 -> cantidadPuntos += 1;
        default -> cantidadPuntos += 0;
    }
}
    public F1 subirCategoria(String nombrePatrocinador) {
        if(superLicencia){    
        return new F1(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos, nombrePatrocinador);
        }else{
            System.out.println(("No puede subir a F1 sin superlicencia"));
            return null;
        }
    }
    public String toString() {
        return super.toString() + " | Superlicencia: " + (superLicencia ? "Sí" : "No");
    }
}

File: Formula1/bin/Formula1/PHP/F2.php
<?php
namespace Formula1;

class F2 extends Monoplaza {
    private $superLicencia;

     ////////CONSTRUCTOR////////
    public function __construct(string $nombrePiloto, string $nacionalidad, int $numeroPlaza, string $escuderia, int $cantidadPuntos, bool $superLicencia) {
        parent::__construct($nombrePiloto, $nacionalidad, $numeroPlaza, $escuderia, $cantidadPuntos);
        $this->superLicencia = $superLicencia;
    }

     // GETTERS Y SETTERS ////
    public function isSuperLicencia(){
        return $this->superLicencia;
    }

    public function setSuperLicencia($valor){
    $this->superLicencia = $valor;
    }

     // METODOS ///
    public function posicionValida($posicion){
        if ($posicion < 1) {
        return false;
    }
    if ($posicion > 24) {
        return false;
    }
    return true;
    }

    public function otorgarPuntos($posicion,$vueltaRapida){
        if ($posicion < 1 || $posicion > 24) {
        return false;
        }
        switch ($posicion) {
        case 1:
            $this->cantidadPuntos += 10;
            break;
        case 2:
            $this->cantidadPuntos += 8;
            break;
        case 3:
            $this->cantidadPuntos += 7;
            break;
        case 4:
            $this->cantidadPuntos += 6;
            break;
        case 5:
            $this->cantidadPuntos += 5;
            break;
        case 6:
            $this->cantidadPuntos += 4;
            break;
        case 7:
            $this->cantidadPuntos += 3;
            break;
        case 8:
            $this->cantidadPuntos += 2;
            break;
        case 9:
            $this->cantidadPuntos += 1;
            break;
        default:
            break;
        }
    }

    public function subirCategoria($nombrePatrocinador){
        if ($this->superLicencia) {
            return new F1($this->nombrePiloto, $this->nacionalidad, $this->numeroPlaza, $this->escuderia, $this->cantidadPuntos, $nombrePatrocinador);
        } else {
            echo "No puede subir a F1 sin superlicencia.\n";
            return null;
        }
    }

    ///TOSTRING///
    public function __toString(): string {
    return parent::__toString() . "Superlicencia:" . ($this->superLicencia ? "Sí" : "No");
}
}